jquery-deparam/index.d.ts

```diff
@@ -0,0 +1,9 @@
+// Type definitions for jquery-deparam v1.0.0
+// Project: https://github.com/AceMetrix/jquery-deparam
+// Definitions by: Pat Sissons <http://github.com/patsissons>
+
+export as namespace deparam;
+
+export = deparam;
+
+declare function deparam(params: string, coerce?: boolean): any;
```

Comment:
Nevermind, it is harder than that...

The "first order" return will be an object with string index, but the values can be `string | string[] | <<recurse>>`.